### PR TITLE
Reduce cargo doc warnings

### PR DESCRIPTION
**Pull request by gilescope**
Fixes some of the cargo doc warnings. *No code changes*

### CODE REVIEW

**Comment by gilescope** [comment]
Let me know if I have got the process wrong in any way.

File: client/network/src/config.rs [hunk 1]
```diff
@@ -599,8 +599,7 @@ pub enum TransportConfig {
 
 		/// If true, allow connecting to private IPv4 addresses (as defined in
 		/// [RFC1918](https://tools.ietf.org/html/rfc1918)). Irrelevant for addresses that have
-		/// been passed in [`NetworkConfiguration::reserved_nodes`] or
-		/// [`NetworkConfiguration::boot_nodes`].
+		/// been passed in [`NetworkConfiguration::boot_nodes`].
 		allow_private_ipv4: bool,
```
Comment (gilescope) [inline client/network/src/config.rs:603]: reserved_nodes don't seem to be a thing any more.

File: primitives/npos-elections/compact/src/lib.rs [hunk 2]
```diff
@@ -419,7 +419,9 @@ fn check_compact_attr(input: ParseStream) -> Result<bool> {
 	}
 }
 
+/// ```ignore
```
Comment (bkchr) [inline primitives/npos-elections/compact/src/lib.rs:422]:
```suggestion
```

File: primitives/npos-elections/compact/src/lib.rs [hunk 3]
```diff
@@ -419,7 +419,9 @@ fn check_compact_attr(input: ParseStream) -> Result<bool> {
...
 /// #[compact] pub struct CompactName::<VoterIndex = u32, TargetIndex = u32, Accuracy = u32>()
```
Comment (bkchr) [inline primitives/npos-elections/compact/src/lib.rs:423]:
```suggestion
/// `#[compact] pub struct CompactName::<VoterIndex = u32, TargetIndex = u32, Accuracy = u32>()`
```

File: primitives/npos-elections/compact/src/lib.rs [hunk 4]
```diff
@@ -419,7 +419,9 @@ fn check_compact_attr(input: ParseStream) -> Result<bool> {
...
+/// ```
```
Comment (bkchr) [inline primitives/npos-elections/compact/src/lib.rs:424]:
```suggestion
```

**Review by bkchr — APPROVED** [review]

**Review by adoerr — APPROVED** [review]

**Pull request merged**